compiler/plugins/target/AMD-AIE/aie/AMDAIEObjectFifoStatefulTransform.cpp

```diff
@@ -1092,16 +1049,14 @@ struct AMDAIEObjectFifoStatefulTransformPass : mlir::OperationPass<DeviceOp> {
     }
 
     // Remove old ops
-    SetVector<Operation *> opsToErase;
+    IRRewriter rewriter(&getContext());
     device.walk([&](Operation *op) {
       if (isa<ObjectFifoCreateOp, ObjectFifoLinkOp, ObjectFifoAcquireOp,
-              ObjectFifoSubviewAccessOp, ObjectFifoReleaseOp>(op))
-        opsToErase.insert(op);
+              ObjectFifoSubviewAccessOp, ObjectFifoReleaseOp>(op)) {
+        op->dropAllUses();
+        rewriter.eraseOp(op);
+      }
     });
-    topologicalSort(opsToErase);
```

Comment:
I added this actually (in mlir-aie) because if you don't do this in toposort order and it fails (and the pass doesn't complete) you can leave the IR in an invalid state thereby preventing debug dumps. But I guess `dropAllUses` takes care of the same problem.

Reply:
Yeah, this is not too important, I was just trying to get rid of some stuff. Soon none of this should be needed anyway.